src/utils.ts

```diff
@@ -1097,6 +1098,17 @@ export function isSuperset(set: Set<any> | any[], subset: Set<any> | any[]): boo
   return true;
 }
 
+/**
+ * Checks if the document is a Hello request
+ * @internal
+ */
+export function isHello(doc: Document): boolean {
+  if (doc[LEGACY_HELLO_COMMAND] || doc.hello) {
```

Comment:
nit: we could inline this if we wanted to do it in one line, eg `return (doc[LEGACY_HELLO_COMMAND] || doc.hello) ? true : false`

Reply:
@durran Fixed. Can you take another look?

Reply:
Looks great, thanks!